source/exe/main_common.cc

```diff
@@ -55,7 +55,11 @@ MainCommonBase::MainCommonBase(const Server::Options& options, Event::TimeSystem
   // Enable core dumps as early as possible.
   if (options_.coreDumpEnabled()) {
     auto ret = platform_impl_.enableCoreDump();
-    ENVOY_LOG_MISC(info, "core dump enabled: {}", ret);
+    if (ret) {
+      ENVOY_LOG_MISC(info, "core dump enabled: {}", ret);
+    } else {
+      ENVOY_LOG_MISC(warn, "core dump enabled: {}", ret);
```

Comment:
nit: I don't think you need to log ret now. Also, can you cover both of these cases in tests? It should be pretty easy.

Reply:
Sure np